src/compiler/moduleNameResolver.ts

```diff
@@ -773,17 +774,14 @@ namespace ts {
                 if (!resolved) return undefined;
 
                 let resolvedValue = resolved.value;
-                let originalPath: string | undefined;
-                if (!compilerOptions.preserveSymlinks && resolvedValue) {
-                    originalPath = resolvedValue.path;
+                if (!compilerOptions.preserveSymlinks && resolvedValue && !resolvedValue.originalPath) {

```

Comment:
Hmm we need better way to identify that its from cache and no need to calculate originalPath since when its not symbollink and its coming from cache we shouldnt need to do realPath logic here. Considering this involves more changes than previously expected, do you mind if i take it over?

Reply:
I don't mind you taking over.
I think this could easily be fixed by always setting `originalPath` to the original path regardless of the result of `realpath`. If `originalPath` is present it's from the cache, otherwise resolve the path using realpath